File: software/examples/javascript/ExampleThreshold.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your Barometer Bricklet 2.0

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var b = new Tinkerforge.BrickletBarometerV2(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Configure threshold for air pressure "greater than 1025 mbar"
        // with a debounce period of 1s (1000ms)
        b.setAirPressureCallbackConfiguration(1000, false, '>', 1025*1000, 0);
    }
);

// Register air pressure callback
b.on(Tinkerforge.BrickletBarometerV2.CALLBACK_AIR_PRESSURE,
    // Callback function for air pressure callback
    function (airPressure) {
        console.log('Air Pressure: ' + airPressure/1000.0 + ' mbar');
        console.log('Enjoy the potentially good weather!');
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/ruby/example_threshold.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_barometer_v2'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your Barometer Bricklet 2.0

ipcon = IPConnection.new # Create IP connection
b = BrickletBarometerV2.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Register air pressure callback
b.register_callback(BrickletBarometerV2::CALLBACK_AIR_PRESSURE) do |air_pressure|
  puts "Air Pressure: #{air_pressure/1000.0} mbar"
  puts 'Enjoy the potentially good weather!'
end

# Configure threshold for air pressure "greater than 1025 mbar"
# with a debounce period of 1s (1000ms)
b.set_air_pressure_callback_configuration 1000, false, '>', 1025*1000, 0

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect
